lib/improver/blending/weighted_blend.py

```diff
@@ -37,10 +37,123 @@
+            Cubes that have had their forecast_reference_time unified.
+
+    Raises:
+        ValueError: if forecast_reference_time is a dimension coordinate
+    """
+    if isinstance(cubes, iris.cube.Cube):
+        cubes = iris.cube.CubeList([cubes])
+
+    result_cubes = iris.cube.CubeList([])
+    for cube in cubes:
+        frt_units = cube.coord('forecast_reference_time').units
+        frt_points = [frt_units.date2num(cycletime)]
+        frt_coord = build_coordinate(
+            frt_points, standard_name="forecast_reference_time", bounds=None,
+            template_coord=cube.coord('forecast_reference_time'),
+            data_type=np.float64)
+        cube.remove_coord("forecast_reference_time")
+        cube.add_aux_coord(frt_coord, data_dims=None)
+
+        # If a forecast period coordinate already exists on a cube, replace
+        # this coordinate, otherwise create a new coordinate.
+        fp_units = "seconds"
+        if cube.coords("forecast_period"):
+            fp_units = cube.coord("forecast_period").units
+            cube.remove_coord("forecast_period")
+        fp_coord = forecast_period_coord(
+            cube, force_lead_time_calculation=True, result_units=fp_units)
+        cube.add_aux_coord(fp_coord, data_dims=cube.coord_dims("time"))
+        result_cubes.append(cube)
+    return result_cubes
+
+
+def rationalise_blend_time_coords(
+        cubelist, blend_coord, cycletime=None, weighting_coord=None):
+    """
+    Updates time coordinates on unmerged input cubes before blending depending
+    on the coordinate over which the blend will be performed.  Modifies cubes
+    in place.
+
+    If blend_coord is forecast_reference_time, ensures the cube has
+    a forecast_period dimension.  If weighting_coord is forecast_period,
+    equalises forecast_reference_time on each cube before blending.
+
+    Args:
+        cubelist (iris.cube.CubeList):
+            List of cubes containing data to be blended
+        blend_coord (str):
+            Name of coordinate over which the blend will be performed
+
+    Kwargs:
+        cycletime (str or None):
+            The cycletime in a YYYYMMDDTHHMMZ format e.g. 20171122T0100Z
+        weighting_coord (str or None):
+            The coordinate across which weights will be scaled in a
+            multi-model blend
+
+    Raises:
+        ValueError: if forecast_reference_time (to be unified) is a
+            dimension coordinate
+    """
+    if "forecast_reference_time" in blend_coord:
+        for cube in cubelist:
+            coord_names = [x.name() for x in cube.coords()]
+            if "forecast_period" not in coord_names:
+                forecast_period = forecast_period_coord(cube)
+                cube.add_aux_coord(forecast_period, data_dims=None)
+
+    # if blending models using weights by forecast period, set forecast
+    # reference times to current cycle time
+    if "model" in blend_coord and "forecast_period" in weighting_coord:
```

Comment:
Is this true only if blend_coord is "model" or is it true if you were doing any other kind of blending? I know we only have the model use case at the moment.

Reply:
As far as I know the only use cases are model blending (`if "model" in blend_coord`, which covers things like "model_configuration" or similar coordinate names) and cycle blending (`blend_coord == 'forecast_reference_time'`).  If we're blending over cycles, I can't think why we'd want to equalise the cycle times - and in fact I think that would be dangerous as we might get duplicate cubes or lose weighting factors by forecast period.

Reply:
Yep, that's fine